src/coreclr/tests/build.proj

```diff
@@ -52,6 +52,19 @@
     <Exec Command="$(DotnetRestoreCommand)"/>
   </Target>
 
+  <Target Name="GenerateTestAssemblyList">
+	<PropertyGroup>
+		<TestAssemblyListPath>$(RepoRoot)/artifacts/tests/coreclr/$(OSPlatformConfig)/</TestAssemblyListPath>
+	</PropertyGroup>	
+	<Message Importance="High" Text="Generating test_assemblies.txt:  $(TestAssemblyListPath)" />
+	<Exec ConsoleToMSBuild="true" Command="find $(TestAssemblyListPath) -name *.dll > $(TestAssemblyListPath)/test_assemblies.txt" />
+  </Target>
+
+  <Target Name="BuildWasmUtilities" AfterTargets="Build" Condition="'$(TargetArchitecture)' == 'wasm' " >
+	<CallTarget Targets="GenerateTestAssemblyList" />
+	<MSBuild Projects="$(RepoRoot)/src/mono/wasm/wasm-test-driver/WasmTestDriver.csproj;$(RepoRoot)/src/mono/wasm/wasm-test-runner/WasmTestRunner.csproj" Targets="Restore;Build" /> 
```

Comment:
I would recommend doing restore and build as separate steps, doing it on the same msbuild node causes issue because the project is not re-evaluated after restore step and restore generates some files that need to be imported for build step, which will only happen correctly if the project is re-evaluated.